src/main/webapp/app/exercises/shared/exercise-detail-common-actions/non-programming-exercise-detail-common-actions.component.html

```diff
@@ -51,6 +51,12 @@
             <span class="d-none d-md-inline" jhiTranslate="entity.action.exampleSubmissions">Example Submissions</span>
         </a>
 
+        <!-- Cluster Statistics  -->
+        <a *ngIf="exercise.isAtLeastInstructor && exercise.type == 'text'" [routerLink]="baseResource + 'text-cluster-statistics'" class="btn btn-success btn-sm me-1 mb-2">
+            <fa-icon [icon]="'chart-bar'"></fa-icon>
+            <span class="d-none d-md-inline" jhiTranslate="artemisApp.textExercise.clusterStatistics.title">Cluster Statistics</span>
+        </a>
+
```

Comment:
what do you think about showing this button only if there are some clusters found for this exercise? so: `*ngIf="clusters.length > 0"` in addition to checking where it is e text exercise? Then you would not need 
`<span *ngIf="clusters.length == 0" jhiTranslate="artemisApp.textExercise.clusterStatistics.noClustersFound"><b>No text clusters found for this exercise</b></span>`
within the component. As a user I would prefer only seeing the button Cluster Analysis, if there actually is an analysis.
I don't know if you can easily refactor the code, so that you know the length of the cluster before initializing `ClusterStatisticsComponent`.

Reply:
That's one good suggestion 👍 . I see multiple problems with it however. 
First, in order to check clusters.length,  the clusters first need to be fetched. For that, a lot of logic needs to be moved and re-organized so that it is available in that component instead. Would rather seperate cluster fetching to clusters component instead (separation of concerns). Second, the clusters may contain lot of data for large courses, which means only for this fetching, the whole page with the other buttons has to wait. As Cluster Analysis Page is not as frequently used as the other common buttons in Text Exercise (edit, delete, etc), they should not have to wait extra. Overall, quite a lot of overall effort for little benefit.